platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -1385,19 +1382,18 @@ private void deserializePathToDirtyLeafIndex(final Map<Long, Mutation<K>> map, f
      * 		If something fails.
      */
     private void serializeKeyToDirtyLeafIndex(
-            final Map<K, Mutation<VirtualLeafRecord<K, V>>> map, final SerializableDataOutputStream out)
+            final Map<K, Mutation<K, VirtualLeafRecord<K, V>>> map, final SerializableDataOutputStream out)
             throws IOException {
         assert snapshot.get() : "Only snapshots can be serialized";
         out.writeInt(map.size());
-        for (final Map.Entry<K, Mutation<VirtualLeafRecord<K, V>>> entry : map.entrySet()) {
-            final Mutation<VirtualLeafRecord<K, V>> mutation = entry.getValue();
+        for (final Map.Entry<K, Mutation<K, VirtualLeafRecord<K, V>>> entry : map.entrySet()) {
+            final Mutation<K, VirtualLeafRecord<K, V>> mutation = entry.getValue();
             assert mutation != null : "Mutations cannot be null in a snapshot";
             assert mutation.version <= this.fastCopyVersion.get()
                     : "Trying to serialize keyToDirtyLeafIndex with a version ahead";
 
             final VirtualLeafRecord<K, V> leaf = mutation.value;
             out.writeSerializable(leaf, false);
-            out.writeSerializable(leaf.getHash(), true);
```

Comment:
Same compatibility concerns here.

Reply:
The compatibility is taken care of now:
`deserializeKeyToDirtyLeafIndex`

```
            if (version == ClassVersion.ORIGINAL) {
                // skip path
                in.readLong();
            }
```

Reply:
Fixed.